server/src/main/java/org/opensearch/search/aggregations/bucket/composite/GlobalOrdinalValuesSource.java

```diff
@@ -171,17 +169,39 @@ LeafBucketCollector getLeafCollector(LeafReaderContext context, LeafBucketCollec
         if (lookup == null) {
             initLookup(dvs);
         }
+
+        // unwrapSingleton() returns non-null only if the field is single-valued
+        final SortedDocValues singleton = DocValues.unwrapSingleton(dvs);
```

Comment:
I'm afraid there is little optimization, it only save the step:
```
int count = dvs.docValueCount();
```

Reply:
@kkewwei - While we are single invocation, the `collect` is invoked for every single document. So, if we are collecting million documents, we save that many invocations of `dvs.docValueCount()`. Probably due to which, it is noticeable in the benchmark as well.